homeassistant/components/control4/light.py

```diff
@@ -0,0 +1,224 @@
+                    item_device_name = parent_item["name"]
+                    item_model = parent_item["model"]
+            async_add_entities(
+                [
+                    Control4Light(
+                        hass,
+                        entry,
+                        item_coordinator,
+                        item_name,
+                        item_id,
+                        item_device_name,
+                        item_manufacturer,
+                        item_model,
+                        item_parent_id,
+                        item_is_dimmer,
+                        light_transition_time,
+                        light_cold_start_transition_time,
+                    )
+                ],
+                True,
+            )
+
+
+class Control4Light(Control4Entity, LightEntity):
+    """Control4 light entity."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+        coordinator: DataUpdateCoordinator,
+        name: str,
+        idx: int,
+        device_name: str,
+        device_manufacturer: str,
+        device_model: str,
+        device_id: int,
+        is_dimmer: bool,
+        light_transition_time: int,
+        light_cold_start_transition_time: int,
+    ):
+        """Initialize Control4 light entity."""
+        super().__init__(
+            hass,
+            entry,
+            coordinator,
+            name,
+            idx,
+            device_name,
+            device_manufacturer,
+            device_model,
+            device_id,
+        )
+        self._is_dimmer = is_dimmer
+        self._transition_time = light_transition_time
+        self._cold_start_transition_time = light_cold_start_transition_time
+        self._c4_light = None
+
+    async def async_added_to_hass(self):
+        """When entity is added to hass."""
+        await super().async_added_to_hass()
+        self._c4_light = C4Light(self.director, self._idx)
+
+    @property
+    def is_on(self):
+        """Return whether this light is on or off."""
+        return bool(self._coordinator.data[self._idx]["value"] > 0)
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 0..255."""
+        if self._is_dimmer:
+            return self._coordinator.data[self._idx]["value"] * 2.55
+        return None
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        flags = 0
+        if self._is_dimmer:
+            flags |= SUPPORT_BRIGHTNESS | SUPPORT_TRANSITION
+        return flags
+
+    async def async_turn_on(self, **kwargs) -> None:
+        """Turn the entity on."""
+        if self._is_dimmer:
+            if ATTR_TRANSITION in kwargs:
+                transition_length = kwargs[ATTR_TRANSITION] * 1000
+            elif self.brightness == 0:
+                transition_length = self._cold_start_transition_time * 1000
```

Comment:
I suggest we remove this option from this PR and explore how to add this feature to the light integration.